### DISCUSSION TITLE
Suspense Fallback failing to render / appear on screen

### DISCUSSION THREAD

**Discussion by yaq-cc** [solidjs/solid#1597] [Q&A]
### Describe the bug

Hi team, 

This is WRT to Suspense's fallback property which doesn't seem to resolve at all in this use case.  Show seems much easier to use that Suspense - I think SolidJS Suspense needs more use case examples as the only ones that worked for me were with lazily loaded components.  

### Your Example Website or App

https://playground.solidjs.com/anonymous/66efe535-fec0-47a9-a195-07490d3bc279

### Steps to Reproduce the Bug or Issue

Fallback component: 
```javascript
const FallbackComponent = () => {
	console.log(`Fallback Component...`)
	return (<Box>
				<p>Loading...</p>
				<Button>Hello World!</Button>	
	</Box>)

}
```
App Component: 
```javascript
export default function App() {
	const [data, setData] = createSignal<Data[]>();
	const LazyFor = lazy(async () => {
		getData()
			.then((res) => {
				setData(res);
				console.log(res);
			})
		return {
			default: () => (<For each={data()}>{(record, index) => {
				return (<p><ChevronRight/>{record.name}</p>)
			}}</For>)
		}
	})
	return (<Box>
		<Typography>
			<Suspense fallback={<FallbackComponent />}>
				<LazyFor />
			</Suspense>
		</Typography>
	</Box>)
}```


### Expected behavior

As a user, I expected to see a paragraph that says "loading..." and a Hello World button.  What I got as a user was a blank screen that was updated with my component when the promise resolves.  



### Screenshots or Videos

a

### Platform

- OS: [Debian Linux]
- Browser: [Chrome]
- Version: [110.0.5481.178 ]


### Additional context

_No response_

**Comment by yaq-cc** [comment]
Is suspense only for lazily loaded components?  Should I be using Show in this fashion instead?  I can't seem to get the fallback component to render.  I can see in the console the component is, in fact, invoked - but I can't see it rendered on the page.

https://playground.solidjs.com/anonymous/8ddb6442-0922-4036-a4ea-cdca6eb186c7

```typescript
import { render } from "solid-js/web"
import { ParentProps, For, Show, createSignal } from "solid-js";
import { source, Data } from "./tab1"

function getData(timeout: number = 1000) {
    return new Promise<Data[]>((resolve, reject) => {
        setTimeout(() => {
            resolve(source);
        }, timeout)
    })
}

function FallbackComponent() {
	console.log("FallbackComponent invoked");
	return (<>
		<p>Loading...</p>
	</>)
}

function ForComponent(props: ParentProps<{each: Data[]}>) {
	return <For each={props.each}>{(item) => {
		return <p>{item.name}</p>
	}}</For>
}

export default function App() {
	const [data, setData] = createSignal<Data[]>();
	
	getData()
	.then(res => {
		setData(res);
	})
	
	return (<>
		<Show when={data()} fallback={<FallbackComponent/>}>
			<ForComponent each={data() as Data[]}/>
		</Show>
	</>)
}

render(() => <App />, document.getElementById("app")!);
```

**Comment by ryansolid** [comment]
Generally you need to use `createResource` which `lazy` uses under the hood. After removing the SUID stuff that wasn't working in the playground I could see the problem was the lazy component resolved immediately hence no fallback. `await`ing the data request works: https://playground.solidjs.com/anonymous/916a925b-968d-41e0-a8a2-b7fa3530edba

That being said I'd probably do this instead:
```js
function App() {
  const [data] = createResource<Data[]>(() => getData());
  return (
    <div>
      <Suspense fallback={<FallbackComponent />}>
        <For each={data()}>
          {(record, index) => {
            return <p>{record.name}</p>;
          }}
        </For>
      </Suspense>
    </div>
  );
}
```
Full example: https://playground.solidjs.com/anonymous/0056f3d9-1b5e-4580-a6cb-8eb1dd5ac58b

SuspenseList and Transition Tutorial show some resource examples: https://www.solidjs.com/tutorial/async_suspense_list
